source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -266,6 +267,16 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
     idle_timeout_ = absl::nullopt;
   }
 
+  // If the path normalization options has been set.
+  if (config.has_path_normalization_options()) {
+    normalize_path_ = false;
+    merge_slashes_ = false;
+    forwarding_path_transformer_ = std::make_unique<Http::PathTransformer>(
```

Comment:
You can use these transformers for the old normalization config as well. The filter_path_transformer_ will be no-op and forwarding_path_transformer_ will have rfc normalization and merge slashes depending on the `normalize_path` and `merge_slashes` values.
In this case you will not need to have two different ways to transform the path in the `ConnectionManagerUtility::maybeNormalizePath`

Reply:
Thanks , have cleaned up.